internal/build/imgsrc/ensure_builder.go

```diff
@@ -409,6 +410,8 @@ func restartBuilderMachine(ctx context.Context, builderMachine *fly.Machine) err
 
 	flapsClient := flapsutil.ClientFromContext(ctx)
 
+	mach.WaitForAnyMachineState(ctx, builderMachine, []string{"started", "stopped"}, 60*time.Second, nil)
```

Comment:
Should we be checking the return values? Or are we restarting the machine even if it's already in `started` state which seems to be the desired state?

Reply:
yes we absolutely should. thank you for pointing this out :)